stf.core/scripts/stf/Commands.pm

```diff
@@ -691,38 +691,49 @@ sub kill_process {
 	my $ppid = $process->{pid};
 
 	$process->{killed} = $FALSE;
-        
-	unless ($process->poll()) {
-		info("  o Process $process->{uid} pid $ppid is not running");
-		return;
-	}
 
-	info("  o Process $process->{uid} pid $ppid stop()");
-	$process->stop();
+	my $attempt = 1;
 
-	if($process->poll()) {
-		info("  o Process $process->{uid} pid $ppid terminate()");
-		$process->terminate();
-	}
+	while($attempt < 6) {
+		info("  o Process clean up attempt $attempt for $process->{uid} pid $ppid");
+		unless ($process->poll()) {
+			info("  o Process $process->{uid} pid $ppid is not running");
+			return;
```

Comment:
Having a return here, in particular inside the loop, seems wrong. Possibly there needs to be some time after calling stop() before checking if the process is still running. The process could be found stopped after looping around. The later code echos any remaining output and returns a 0 or 1 code.

Reply:
The later code also does `$process->{killed} = $TRUE;` which may not be desirable if stop() was never called.

Reply:
> Having a return here, in particular inside the loop, seems wrong.

Thanks for spotting! Moved the while loop below that check (of whether the process is already dead). 

 >  Possibly there needs to be some time after calling stop() before checking if the process is still running

There already exists a 10 second wait [inside the stop() subroutine](https://github.com/adoptium/STF/blob/master/stf.core/scripts/stf/ProcessMgmt/UNIX.pm#L295) - which is why I didn't introduce any additional wait after invocation of stop().